libraries/src/Table/Content.php

```diff
@@ -252,31 +252,31 @@ public function check()
 			$this->hits = 0;
 		}
 
-		// Set publish_up to null date if not set
+		// Set publish_up to null if not set
 		if (!$this->publish_up)
 		{
-			$this->publish_up = $this->_db->getNullDate();
+			$this->publish_up = null;
 		}
 
-		// Set publish_down to null date if not set
+		// Set publish_down to null if not set
 		if (!$this->publish_down)
 		{
-			$this->publish_down = $this->_db->getNullDate();
+			$this->publish_down = null;
 		}
 
 		// Check the publish down date is not earlier than publish up.
-		if ($this->publish_down < $this->publish_up && $this->publish_down > $this->_db->getNullDate())
+		if (!is_null($this->publish_up) && !is_null($this->publish_down) && $this->publish_down < $this->publish_up)
 		{
 			// Swap the dates.
 			$temp = $this->publish_up;
 			$this->publish_up = $this->publish_down;
 			$this->publish_down = $temp;
 		}
 
-		// Set modified to null date if not set
+		// Set modified to created date if not set
 		if (!$this->modified)
 		{
-			$this->modified = $this->_db->getNullDate();
+			$this->modified = $this->created;
```

Comment:
This won't work - `$this->created` is only set in the store method :/ We'll have to move this down there (somewhere around line 347) - or probably better move that check into this method - as it is a check

Reply:
> or probably better move that check into this method - as it is a check

Sorry, don't understand which check in which method.

Reply:
You mean moving the complete `if ($this->id)` stuff including the `else` part from method `store()` up to method `check()`? Would that work?

Reply:
First way is easier and I think I've already seen that somewhere else.